### DISCUSSION TITLE
How see UI during execution of Jasmine spec with Spectacular?

### DISCUSSION THREAD

**Discussion by meirka** [ngworker/ngworker#34] [Q&A]
With Angular TestBed I can see UI (shows up) in Jasmine. It is particularly useful to me when I debug and can see current state of UI and/or when test fails, again I can see last state of UI (which helps to understand what's going on). Granted I have teardown after each test disabled in test.ts file. Is there a way to make Spectacular show UI? If not, what are alternatives to viewing UI as test executes? I disabled teardown in the project ( https://github.com/meirka/angular-unit-test-navigation ), now you can run Jasmine and see UI (for illustrational purposes).

Thank you.

**Comment by LayZeeDK** [comment]
Spectacular's Feature testing API and Pipe testing API use Angular's TestBed behind the scenes. If you're using Karma, use its debug and pause features to inspect the DOM being rendered. You shouldn't disable Angular testing module teardown as it increases correctness, consistency, and speed.
